Source/SwiftLintFramework/Rules/Lint/AccessibilityLabelForImageRule.swift

```diff
@@ -0,0 +1,233 @@
+        if substructure.contains(where: { $0.hasAccessibilityHiddenModifier(in: file) }) {
+            return true
+        }
+
+        return false
+    }
+
+    /// Whether or not the dictionary represents a SwiftUI View with an `accesibilityLabel(_:)`
+    /// or `accessibility(label:)` modifier.
+    var hasAccessibilityLabelModifier: Bool {
+        guard expressionKind == .call, let name: String = name else {
+            return false
+        }
+
+        // check for iOS 14+ version of modifier
+        if name.hasSuffix("accessibilityLabel") {
+            return true
+        }
+
+        // check for iOS 13 version of modifier
+        if name.hasSuffix("accessibility"), enclosedArguments.contains(where: { $0.name == "label" }) {
+            return true
+        }
+
+        // recursively check substructure
+        // SwiftUI literal Views with modifiers will have a SourceKittenDictionary structure like:
+        // Image("myImage").resizable().accessibilityLabel(Text("Label")).frame
+        //     '--> Image("myImage").resizable().accessibilityLabel
+        if substructure.contains(where: { $0.hasAccessibilityLabelModifier }) {
+            return true
+        }
+
+        return false
+    }
+
+    /// Whether or not the dictionary represents a SwiftUI View with an `accessibilityElement()` or
+    /// `accessibilityElement(children: .ignore)` modifier (`.ignore` is the default parameter value).
+    func hasAccessibilityElementChildrenIgnoreModifier(in file: SwiftLintFile) -> Bool {
+        guard expressionKind == .call, let name: String = name else {
+            return false
+        }
+
+        // check for modifier
+        if name.hasSuffix("accessibilityElement") {
+            if enclosedArguments.isEmpty {
+                return true
+            }
+
+            if let childrenArg: SourceKittenDictionary = enclosedArguments.first(where: { $0.name == "children" }),
+                childrenArg.getArgumentValue(in: file) == ".ignore" {
+                return true
+            }
+        }
+
+        // recursively check substructure
+        // SwiftUI literal Views with modifiers will have a SourceKittenDictionary structure like:
+        // VStack { ... }.accessibilityElement().padding
+        //     '--> VStack { ... }.accessibilityElement
+        if substructure.contains(where: { $0.hasAccessibilityElementChildrenIgnoreModifier(in: file) }) {
+            return true
+        }
+
+        return false
+    }
+
+    /// Helper to get the value of an argument
+    func getArgumentValue(in file: SwiftLintFile) -> String? {
+        guard expressionKind == .argument, let bodyByteRange: ByteRange = bodyByteRange else {
+            return nil
+        }
+
+        return file.stringView.substringWithByteRange(bodyByteRange)
+    }
+
+    /// Helper to get the value of a single unnamed argument to a function call
+    func getSingleUnnamedArgumentValue(in file: SwiftLintFile) -> String? {
```

Comment:
It may just that this is my first time working with SourceKitten, but to me the function name is a lot more clear than seeing an inline check that this is a `.call` expression with a certain value for `file.stringView.substringWithByteRange(bodyByteRange)`. Happy to make the change anyways if you feel strongly about it, but this is probably the one comment I'd push back on a bit. Could also see something like this potentially being useful for future SwiftUI rules (especially other accessibility rules) even if I'm only using it once in this rule.

Reply:
Fine for me if you keep it. I see your point.